pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -157,6 +169,13 @@ func (a *AgentClusterInstall) Load(f asset.FileFetcher) (bool, error) {
 		err = errors.Wrapf(err, "failed to unmarshal %s", agentClusterInstallFilename)
 		return false, err
 	}
+
+	if agentClusterInstall.Spec.Networking.NetworkType == "" {
+		defaultInstallConfig := &types.InstallConfig{}
+		agentClusterInstall.Spec.Networking.NetworkType = a.getDefaultNetworkType(defaultInstallConfig)
+		logrus.Infof("NetworkType is not specified in AgentClusterInstall. Defaulting to %s.", agentClusterInstall.Spec.Networking.NetworkType)
+	}
+
```

Comment:
This block seems to be logically duplicated with the previous one, what about having just one "default" method in common to both the `Load()` and `Generate()`?

Reply:
Sure. I will fold the log message into the getDefaultNetworkType function.